src/Elm.elm

```diff
@@ -1156,6 +1156,135 @@ lambda argBaseName argType toExpression =
         }
 
 
+{-| -}
+lambdaBetaReduced : String -> Elm.Annotation.Annotation -> (Expression -> Expression) -> Expression
+lambdaBetaReduced argBaseName argType toExpression =
+    let
+        arg1 =
+            valueWith [] argBaseName argType
+
+        (Compiler.Expression expr) =
+            toExpression arg1
+    in
+    Compiler.Expression
+        { expression =
+            betaReduce <|
+                Exp.LambdaExpression
+                    { args = [ Compiler.nodify (Pattern.VarPattern argBaseName) ]
+                    , expression = Compiler.nodify expr.expression
+                    }
+        , annotation =
+            case expr.annotation of
+                Err err ->
+                    Err err
+
+                Ok return ->
+                    List.foldr
+                        (\ann fnbody ->
+                            Annotation.FunctionTypeAnnotation
+                                (Compiler.nodify ann)
+                                (Compiler.nodify fnbody)
+                        )
+                        return
+                        [ Compiler.getInnerAnnotation argType ]
+                        |> Ok
+        , imports = expr.imports
+        , skip = False
+        }
+
+
+betaReduce : Exp.Expression -> Exp.Expression
+betaReduce e =
+    let
+        extractLastArg arg =
+            case arg of
+                Exp.FunctionOrValue [] n ->
+                    Just n
+
+                Exp.ParenthesizedExpression p ->
+                    extractLastArg <| Compiler.denode p
+
+                _ ->
+                    Nothing
+
+        -- If the list is nonempty, returns a tuple with the beginning of the list and the last element (denoded).
+        popLast : List (Node.Node a) -> Maybe ( List (Node.Node a), a )
+        popLast lst =
+            case List.reverse lst of
+                [] ->
+                    Nothing
+
+                last :: initReverse ->
+                    Just ( List.reverse initReverse, Compiler.denode last )
```

Comment:
This... can definitely be optimized! Is it worth optimizing? (the lists will be _tiny_ 99.99% of the time)